server/controllers/finance/reports/break_even_fee_center/index.js

```diff
@@ -0,0 +1,147 @@
+  params.end_date = new Date(params.end_date);
+
+  data.period = {
+    start_date : params.start_date,
+    end_date : params.end_date,
+    fiscalYearStart : params.fiscalYearStart,
+  };
+
+  _.defaults(params, DEFAULT_PARAMS);
+
+  try {
+    reporting = new ReportManager(TEMPLATE, req.session, params);
+  } catch (e) {
+    next(e);
+    return;
+  }
+
+  const getFeeCenter = `
+    SELECT fc.id, fc.label, fc.is_principal FROM fee_center AS fc ORDER BY fc.label ASC, fc.is_principal DESC
+  `;
+
+  const getFeeCenterReference = `
+    SELECT fc.id, fc.label, fc.is_principal, rf.fee_center_id, rf.account_reference_id, 
+    rf.is_cost, rf.is_variable, rf.is_turnover, ar.abbr
+    FROM fee_center AS fc
+    JOIN reference_fee_center AS rf ON rf.fee_center_id = fc.id
+    JOIN account_reference AS ar ON ar.id = rf.account_reference_id
+    ORDER BY fc.label
+  `;
+
+  const getFeeCenterDistribution = `
+    SELECT fcd.is_cost, fcd.is_variable, fcd.is_turnover, BUID(fcd.row_uuid) AS row_uuid,
+    fca.label AS auxiliary, fcp.label AS principal, SUM(fcd.debit_equiv) AS debit,
+    SUM(fcd.credit_equiv) AS credit, gl.trans_date, fcd.auxiliary_fee_center_id,
+    fcd.principal_fee_center_id
+    FROM fee_center_distribution AS fcd
+    JOIN general_ledger AS gl ON gl.uuid = fcd.row_uuid
+    JOIN fee_center AS fcp ON fcp.id = fcd.principal_fee_center_id
+    JOIN fee_center AS fca ON fca.id = fcd.auxiliary_fee_center_id
+    WHERE DATE(gl.trans_date) >= DATE(?) AND DATE(gl.trans_date) <= DATE(?)
+    GROUP BY fcd.principal_fee_center_id, fcd.auxiliary_fee_center_id;
+  `;
+
+  const getEncounterFeeCenter = `
+    SELECT f.label AS feeCenter, sfc.fee_center_id, count(pv.uuid) AS numberCase 
+    FROM patient_visit AS pv
+    JOIN patient_visit_service AS pvs ON pvs.patient_visit_uuid = pv.uuid
+    JOIN service_fee_center AS sfc ON sfc.service_id = pvs.service_id
+    JOIN fee_center AS f ON f.id = sfc.fee_center_id
+    WHERE DATE(pv.start_date) >= DATE(?) AND DATE(pv.start_date) <= DATE(?)
+    GROUP BY sfc.fee_center_id   
+  `;
+
+  const dbPromises = [
+    db.exec(getFeeCenter),
+    db.exec(getFeeCenterReference),
+    AccountReference.computeAllAccountReference(params.period_id),
+    db.exec(getFeeCenterDistribution, [params.fiscalYearStart, params.end_date]),
+    db.exec(getEncounterFeeCenter, [params.fiscalYearStart, params.end_date]),
+    getDistributionKey.allDistributionKey(),
+  ];
+
+  q.all(dbPromises)
+    .spread((feeCenter, references, accountReferences, dataDistributions, encounters, distributionKey) => {
+      const config = {
+        feeCenter,
+        references,
+        accountReferences,
+        dataDistributions,
+        distributionKey,
+        encounters,
+        includeManual : params.includeManual,
+      };
+      const dataConfigured = setting.configuration(config);
+
+      if (feeCentersChecked.length) {
+        dataConfigured.principal.forEach(item => {
```

Comment:
Looks like you are trying to get `dataConfigured.principalSelected` at the end of this operation. and that is the only reason you need `item.selected`.  I think this can be done more simply like this:
```js
dataConfigured.prinicpalSelected = dataConfigured.principal
  .filter(item => feeCentersChecked.includes(item.id));
```